dev_scripts/generate-release-tasks.py

```diff
@@ -23,45 +29,29 @@ def git_root():
 
 
 def extract_checkboxes(filename):
-    headers = []
-    result = []
+    first_pass = []
+    second_pass = []
 
     with open(filename, "r") as f:
         lines = f.readlines()
 
-    current_level = 0
     for line in lines:
         line = line.rstrip()
+        if line.startswith("#") or line.lstrip().startswith("- [ ]"):
+            first_pass.append(line)
 
-        # If it's a header, store it
-        if line.startswith("#"):
-            # Count number of # to determine header level
-            level = len(line) - len(line.lstrip("#"))
-            if level < current_level or not current_level:
-                headers.extend(["", line, ""])
-                current_level = level
-            elif level > current_level:
-                continue
-            else:
-                headers = ["", line, ""]
-
-        # If it's a checkbox
-        elif "- [ ]" in line or "- [x]" in line or "- [X]" in line:
-            # Print the last header if we haven't already
-            if headers:
-                result.extend(headers)
-                headers = []
-                current_level = 0
+    current_level = 0
+    for line in reversed(first_pass):
+        level = len(line) - len(line.lstrip("#"))
```

Comment:
Since headers shouldn't contain other `#` chars, I think it's safe to use `str.count()`?

```suggestion
        level = line.count('#')
```

Reply:
Hm, the former looks a bit more safer to me, since we effectively count the hashes from the start of the line. Also, well, it works good enough for a dev script :shrug: